api/tests/core/users/test_api.py

```diff
@@ -2147,6 +2201,27 @@ def test_anonymize_beneficiary_user_iris_not_found(self) -> None:
         assert len(sendinblue_testing.sendinblue_requests) == 0
         assert user_to_anonymize.firstName == "user_to_anonymize"
 
+    def test_anonymize_beneficiary_user_with_unprocessed_gdpr_extract(self) -> None:
+        user_beneficiary_to_anonymize = users_factories.BeneficiaryFactory(
+            firstName="user_beneficiary_to_anonymize",
+            age=18,
+            lastConnectionDate=datetime.datetime.utcnow() - relativedelta(years=3, days=1),
+            gdprUserDataExtract=[users_factories.GdprUserDataExtractBeneficiaryFactory()],
+            deposit__expirationDate=datetime.datetime.utcnow() - relativedelta(years=5, days=1),
+        )
+
+        self.import_iris()
+        iris = geography_models.IrisFrance.query.first()
+
+        with mock.patch("pcapi.core.users.api.get_iris_from_address", return_value=iris):
+            users_api.anonymize_beneficiary_users(force=False)
+
+        db.session.refresh(user_beneficiary_to_anonymize)
+
+        assert len(sendinblue_testing.sendinblue_requests) == 0
+        assert user_beneficiary_to_anonymize.firstName == "user_beneficiary_to_anonymize"
+        assert 1 == users_models.GdprUserDataExtract.query.count()
```

Comment:
```suggestion
        assert users_models.GdprUserDataExtract.query.count() == 1
```
tu fais des yoda conditions maintenant ?

Reply:
dans cet ordre à l'époque, pour des raisons de visibilité, conseillé vous l'avez maitre. 
Changé l'ordre je vais 🤣